src/Microsoft.Framework.Logging.NLog/NLogLoggerProvider.cs

```diff
@@ -21,6 +22,17 @@ public ILogger CreateLogger(string name)
             return new Logger(_logFactory.GetLogger(name));
         }
 
+
+        public void Dispose()
+        {
+            if (!_disposed)
+            {
+                _logFactory.Flush();
+                _logFactory.Dispose();
+                _disposed = true;
```

Comment:
set _disposed before doing any work for all of these. It helps prevent concurrency, error & re-execution, and recursion problems.